### PR TITLE
fix: update cwt claim definitions

### LINKED ISSUE #145: Be more elaborate on CBOR/CWT types and definitions
This includes standalone seperate references for each of the claims rather then referencing the JWT based definition.

### PR DESCRIPTION
**Pull request by tplooker**
Fixes #145

### CODE REVIEW

**Review by c2bo — CHANGES_REQUESTED** [review]
We should adjust adjust lines 415-417 as well

File: draft-ietf-oauth-status-list.md [hunk 1]
```diff
@@ -292,10 +292,10 @@ The following content applies to the CWT protected header:
...
-* `65534` (time to live): OPTIONAL. Same definition as `ttl` claim in [](#status-list-token-jwt).
+* `2` (subject): REQUIRED. As generally defined in {{RFC8392}}. The subject claim MUST specify the URI of the Status List Token. The value MUST be equal to that of the `uri` claim contained in the `status_list` claim of the Referenced Token.
+* `6` (issued at): REQUIRED. As generally defined in {{RFC8392}}. The issued at claim MUST specify the time at which the Status List Token was issued.
+* `4` (expiration time): OPTIONAL. As generally defined in {{RFC8392}}. The expiration time claim, if present, MUST specify the time at which the Status List Token is considered expired by its issuer.
+* `65534` (time to live): OPTIONAL. The time to live claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token can be cached by a consumer before a fresh copy SHOULD be retrieved. The value of the claim MUST be a positive number.
 * `65533` (status list): REQUIRED. The status list claim MUST specify the Status List conforming to the rules outlined in [](#status-list-cbor).
```
Comment (paulbastian) [inline draft-ietf-oauth-status-list.md:299]: Should we talk about what CBOR type this is'? (Major Type 0)?

**Comment by paulbastian** [comment]
please reference CBOR major types as we did for Status List object

**Comment by tplooker** [comment]
Update based on feedback.

File: draft-ietf-oauth-status-list.md [hunk 2]
```diff
@@ -260,10 +260,10 @@ The following content applies to the JWT Header:
...
-* `ttl`: OPTIONAL. The `ttl` (time to live) claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token can be cached by a consumer before a fresh copy SHOULD be retrieved. The value of the claim MUST be a positive number.
+* `sub`: REQUIRED. As generally defined in {{RFC7519}}. The `sub` (subject) claim MUST specify the URI of the Status List Token. The value MUST be equal to that of the `uri` claim contained in the `status_list` claim of the Referenced Token.
+* `iat`: REQUIRED. As generally defined in {{RFC7519}}. The `iat` (issued at) claim MUST specify the time at which the Status List Token was issued.
+* `exp`: OPTIONAL. As generally defined in {{RFC7519}}. The `exp` (expiration time) claim, if present, MUST specify the time at which the Status List Token is considered expired by its issuer.
+* `ttl`: OPTIONAL. The `ttl` (time to live) claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token can be cached by a consumer before a fresh copy SHOULD be retrieved. The value of the claim MUST be a positive number encoded in JSON as a number.
```
Comment (c2bo) [inline draft-ietf-oauth-status-list.md:266]: Should we refer to the same rules as exp ?
```suggestion
* `ttl`: OPTIONAL. The `ttl` (time to live) claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token can be cached by a consumer before a fresh copy SHOULD be retrieved. Its value MUST be a positive number containing a NumericDate value as defined in {{RFC7519}}.
```

File: draft-ietf-oauth-status-list.md [hunk 3]
```diff
@@ -292,10 +292,10 @@ The following content applies to the CWT protected header:
...
-* `4` (expiration time): OPTIONAL. Same definition as `exp` claim in [](#status-list-token-jwt).
-* `65534` (time to live): OPTIONAL. Same definition as `ttl` claim in [](#status-list-token-jwt).
+* `2` (subject): REQUIRED. As generally defined in {{RFC8392}}. The subject claim MUST specify the URI of the Status List Token. The value MUST be equal to that of the `uri` claim contained in the `status_list` claim of the Referenced Token.
+* `6` (issued at): REQUIRED. As generally defined in {{RFC8392}}. The issued at claim MUST specify the time at which the Status List Token was issued.
+* `4` (expiration time): OPTIONAL. As generally defined in {{RFC8392}}. The expiration time claim, if present, MUST specify the time at which the Status List Token is considered expired by its issuer.
+* `65534` (time to live): OPTIONAL. The time to live claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token can be cached by a consumer before a fresh copy SHOULD be retrieved. The value of the claim MUST be a positive number encoded in CBOR as an unsigned integer (Major type 0).
```
Comment (c2bo) [inline draft-ietf-oauth-status-list.md:298]: Same as above: Should we re-use the same data type as exp and iat?
```suggestion
* `65534` (time to live): OPTIONAL. The time to live claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token can be cached by a consumer before a fresh copy SHOULD be retrieved. The value of the claim MUST be a NumericDate, as defined in Section 2 of {{RFC8392}}
```

File: draft-ietf-oauth-status-list.md (hunk 2 above)
Comment (paulbastian) [reply draft-ietf-oauth-status-list.md:266]: ttl is a time duration

File: draft-ietf-oauth-status-list.md [hunk 4]
```diff
@@ -413,8 +413,8 @@ The following content applies to the CWT Claims Set:
...
   * `status_list` (status list): REQUIRED when the status list mechanism defined in this specification is used. It has the same definition as the `status_list` claim in [](#referenced-token-jose) but MUST be encoded as a `StatusListInfo` CBOR structure with the following fields:
-    * `idx`: REQUIRED. Same definition as `idx` claim in [](#referenced-token-jose).
-    * `uri`: REQUIRED. Same definition as `uri` claim in [](#referenced-token-jose).
+    * `idx`: REQUIRED. Unsigned integer (Major Type 0) The `idx` (index) claim MUST specify an Integer that represents the index to check for status information in the Status List for the current Referenced Token. The value of `idx` MUST be a non-negative number, containing a value of zero or greater.
+    * `uri`: REQUIRED. Test string (Major Type 3). The `uri` (URI) claim MUST specify a String value that identifies the Status List or Status List Token containing the status information for the Referenced Token. The value of `uri` MUST be a URI conforming to {{RFC3986}}.
```
Comment (c2bo) [inline draft-ietf-oauth-status-list.md:417]:
```suggestion
    * `uri`: REQUIRED. Text string (Major Type 3). The `uri` (URI) claim MUST specify a String value that identifies the Status List or Status List Token containing the status information for the Referenced Token. The value of `uri` MUST be a URI conforming to {{RFC3986}}.
```

**Review by paulbastian — APPROVED** [review]

**Pull request merged**